### ISSUE TITLE
Umbraco scheduled publish issue with trashed nodes

### ISSUE THREAD

**Issue by thomashdk** [umbraco/Umbraco-CMS#10161]
There is an issue with the scheduled publish/unpublished.
If there is an node where an unpublished date ex. is set. and then the node is manuel unpublished before the scheduled.
Then trash the node.
The scheduled task will still be in the database and break/stop all other scheduled tasks.

## Umbraco version
I am seeing this issue on Umbraco version: 8.12.2

Reproduction
------------
- Set an unpublished date on a node.
- Manuel unpublished
- Delete the node

Now the will be an error in the log like this:

`System.NullReferenceException: Objektreferencen er ikke indstillet til en forekomst af et objekt.
   at Umbraco.Core.Persistence.Repositories.Implement.DocumentRepository.IsPathPublished(IContent content) in D:\a\1\s\src\Umbraco.Core\Persistence\Repositories\Implement\DocumentRepository.cs:line 890
   at Umbraco.Core.Services.Implement.ContentService.IsPathPublished(IContent content) in D:\a\1\s\src\Umbraco.Core\Services\Implement\ContentService.cs:line 749
   at Umbraco.Core.Services.Implement.ContentService.StrategyCanPublish(IScope scope, IContent content, Boolean checkPath, IReadOnlyList`1 culturesPublishing, IReadOnlyCollection`1 culturesUnpublishing, EventMessages evtMsgs, ContentSavingEventArgs savingEventArgs, IReadOnlyCollection`1 allLangs) in D:\a\1\s\src\Umbraco.Core\Services\Implement\ContentService.cs:line 2731
   at Umbraco.Core.Services.Implement.ContentService.CommitDocumentChangesInternal(IScope scope, IContent content, ContentSavingEventArgs saveEventArgs, IReadOnlyCollection`1 allLangs, Int32 userId, Boolean raiseEvents, Boolean branchOne, Boolean branchRoot) in D:\a\1\s\src\Umbraco.Core\Services\Implement\ContentService.cs:line 1151
   at Umbraco.Core.Services.Implement.ContentService.PerformScheduledPublishingExpiration(DateTime date, List`1 results, EventMessages evtMsgs, Lazy`1 allLangs) in D:\a\1\s\src\Umbraco.Core\Services\Implement\ContentService.cs:line 1416
   at Umbraco.Core.Services.Implement.ContentService.PerformScheduledPublish(DateTime date) in D:\a\1\s\src\Umbraco.Core\Services\Implement\ContentService.cs:line 1373
   at Umbraco.Web.Scheduling.ScheduledPublishing.PerformRun() in D:\a\1\s\src\Umbraco.Web\Scheduling\ScheduledPublishing.cs:line 68`
Then trash the node.
If you're filing a bug, please describe how to reproduce it. Include as much
relevant information as possible, such as:

### Bug summary

Trashed nodes with a scheduled time should be removed on delete.
If the trashed node a delete from the trash then all works again.

// Thomas

**Comment by emma-hq** [comment]
Hi @thomashdk 

I'm going to have a go at recreating this and if it's all as described I'll tag as a bug. Thanks so much for all the info here. Plenty to work with.

Emma

**Comment by emma-hq** [comment]
Hey @thomashdk 

I'm not able to recreate this on 8.12.2. Here are the steps I've followed:

1. I schedule an unpublish date and time on the Contact node - this is 2 **minutes** in the future
2. I then manually unpublish (before this time is reached)
3. I delete the Contact node
4. I check my log viewer for errors.

I also tried 

1. I schedule an unpublish date and time on the Contact node - this is 2 **days** in the future
2. I then manually unpublish (before this time is reached)
3. I delete the Contact node
4. I check my log viewer for errors.

In both cases, I saw no errors. Am I missing a step?

![image](https://user-images.githubusercontent.com/78740801/116236618-32ba5e00-a757-11eb-9def-aa736698ecf8.png)


Thanks,

Emma
